src/ncar-bufr2nc-fortran/ncio_mod.f90

```diff
@@ -89,8 +89,12 @@ subroutine write_obs(filedate, write_opt, outdir, itim)
             allocate (ichan(xdata(ityp, itim)%nvars))
             ichan(:) = xdata(ityp, itim)%xseninfo_int(:, iv)
             allocate (obserr(xdata(ityp, itim)%nvars))
-            if (geoinst_list(ityp) == 'ahi_himawari8') then
-               call set_ahi_obserr(geoinst_list(ityp), xdata(ityp, itim)%nvars, obserr)
+            if (write_opt == write_nc_radiance_geo) then
```

Comment:
this `write_nc_radiance_geo` seems very specific flag this is just writing assume geostationary and guess this converter is quite specific in what it handles

Reply:
Take a look at line 87 (you will probably have to click on the up arrow on the left to open up the view) where this if statement is entered when either write_nc_radiance or write_nc_radiance_geo are true. I'm guessing that both write_nc_radiance and write_nc_radiance_geo will not both be true, they will either both be false of just one of them be true. When write_nc_radiance is true the corresponding array inst_list should be queried, and when write_nc_radiance_geo is true the corresponding array geoinst_list should be queried. I'm guessing at this since there are no comments explaining what the actual intention of the code is.

The inst_list has something like 18 entries whereas the geoinst_list has only one. The issue is that when write_nc_radiace is true, the  ityp variable is in a loop counting past 1 which then goes out of bounds on the geoinst_list array. The changes I made were to avoid accessing geoinst_list when write_nc_radiance is true.

Reply:
Thanks for the suggestion! I'll add @hailingz as a reviewer.